Create To-Do List

AaryaNale opened the pull request:
This is a simple To-Do List Application: Manage tasks with ease using this Python-based tkinter app. Add, toggle, and view tasks in a user-friendly graphical interface. Stay organized effortlessly.

AaryaNale committed:
Create README.md

AaryaNale committed:
Create To_Do.py

Code

AaryaNale committed:
Create requirements.md

requirements

AaryaNale commented:
Please review and suggest changes to be made, if any.
Thanks.

FerdausPolok commented on To-Do-List/To_Do.py:
```diff
@@ -0,0 +1,61 @@
+listbox.pack(pady=10)
+
+# Create an entry field to add tasks
+entry = tk.Entry(root)
```
I'm not 100% sure but is this the input field we see on our GUI? If yes can we somehow sanitize it?

AaryaNale replied:
Okay, so we dont want it to take empty tasks, right? 
I'll get on working on it

FerdausPolok replied:
Yeap that would be nice.

FerdausPolok reviewed (changes requested):
Hello! Thanks for your contribution here.
One little change request from my end: can we not accept any empty (ie. "    ")? We may show the user a toast message or maybe just freeze the add task button until we have a valid task title inside our input field.
https://github.com/FerdausPolok/dev-mates/assets/68804751/340b4512-2f1e-4c8c-9d8d-423e59f50a81

Also, it's always a good practice to keep our brances separate. I see you did pushed your code to your main branch. Thanks.

AaryaNale committed:
Update To_Do.py

Generates error window on blank/empty tasks

AaryaNale commented:
Improved the code. Do check it out and let me know if something else needs to be added.
Thanks.

FerdausPolok reviewed (approved):
LGTM 🚀
Optional but we can add a placeholder to our input field for better user experience. Like: Add task title here/ Add your task here/ Your task here...

AaryaNale committed:
Update To_Do.py

##Additions:
- Added "Add a task" placeholder to the input textbox
- Now you can add tasks to the list with the 'Add Task' button provided, or simply press the 'Enter' key to do so
- Right click removes a task from the task list
- Added a scroll bar to deal with a large number of tasks
- Improved the paddingof the elements to make the window look more user-friendly
- Wrapped the "Instructions" text

AaryaNale commented:
Please go through the changes @FerdausPolok

AaryaNale commented:
Up to merge maybe? @FerdausPolok
